src/charm.py

```diff
@@ -492,8 +486,11 @@ def _on_config_changed(self, event: EventBase) -> None:
 
             # persist config to file
             self._mysql.write_content_to_file(path=MYSQLD_CONFIG_FILE, content=new_config_content)
-            self.on[f"{self.restart.name}"].acquire_lock.emit()
-            return
+
+            if self._mysql.is_mysqld_running():
+                # restart the service
+                self.on[f"{self.restart.name}"].acquire_lock.emit()
```

Comment:
This is there to remove a typing issue, since `name` its `AnyStr`.
One can `# ignore`, f-stringit or cast with `str()`.
I'm keeping as is - since I'm maturating the refactor restart idea